server/backup_runner.go

```diff
@@ -70,6 +73,90 @@ func (r *BackupRunner) Run(ctx context.Context, wg *sync.WaitGroup) {
 	}
 }
 
+// TODO(dragonly): Make best effort to assure that users could recover to at least RetentionPeriodTs ago.
+// This may require pending deleting expired backup files and binlog files.
+func (r *BackupRunner) purgeExpiredBackupData(ctx context.Context) {
+	backupSettingList, err := r.server.store.FindBackupSetting(ctx, api.BackupSettingFind{})
+	if err != nil {
+		log.Error("Failed to find all the backup settings", zap.Error(err))
+		return
+	}
+
+	log.Debug("Deleting expired snapshot backup files")
+	for _, bs := range backupSettingList {
+		if bs.RetentionPeriodTs == api.BackupRetentionPeriodUnset {
+			continue
+		}
+		backupDir := path.Join(r.server.profile.DataDir, getBackupRelativeDir(bs.DatabaseID))
+		backupFileInfoList, err := ioutil.ReadDir(backupDir)
+		if err != nil {
+			log.Error("Failed to read backup directory", zap.String("path", backupDir), zap.Error(err))
+			continue
+		}
+		for _, backupFileInfo := range backupFileInfoList {
+			expireTime := backupFileInfo.ModTime().Add(time.Duration(bs.RetentionPeriodTs))
+			if time.Now().After(expireTime) {
+				backupFilePath := path.Join(backupDir, backupFileInfo.Name())
+				if err := os.Remove(backupFilePath); err != nil {
```

Comment:
This isn't sufficient to delete the backup. We should look at and respect the backup metadata list.

Reply:
SG.

Reply:
Didn't see a change.

Reply:
This block is unchanged.
I changed to loop through the backup settings (from PG metadata) and check expiration using backup.UpdatedTs instead of file modification time.

Reply:
What I really mean is to update the backup metadata. The status, more specifically.

Reply:
I see. I added a new backup status type DELETED in this PR https://github.com/bytebase/bytebase/pull/2022.
I'll update the backup status to DELETED after that PR is merged.